tests/check_legacy_setup_coverage

```diff
@@ -176,40 +177,37 @@
 	export COVERAGE_CMD="$(command -v coverage) run --include=setup.py -p"
 fi
 
-make -C ${_TEST_ROOT_DIR} -j1 -f Makefile test-reports || EXIT_CODE=3 ;
-printf "%s\n" "Start of Log:" > ${_TEST_ROOT_DIR}/${LOG_FILE} ; wait ;
-printf "%s\n" "" >> ${_TEST_ROOT_DIR}/${LOG_FILE} ; wait ;
-printf "%s\n" "Start of Setup.py Test one-shot:" >> ${_TEST_ROOT_DIR}/${LOG_FILE} ; wait ;
-printf "%s\n" "" >> ${_TEST_ROOT_DIR}/${LOG_FILE} ; wait ;
+make -C "${_TEST_ROOT_DIR}" -j1 -f Makefile test-reports || EXIT_CODE=3 ;
+printf "%s\n" "Start of Log:" > "${_TEST_ROOT_DIR}"/"${LOG_FILE}" ; wait ;
+printf "\n%s\n\n" "Start of Setup.py Test one-shot:" >> "${_TEST_ROOT_DIR}"/"${LOG_FILE}" ; wait ;
 
-{ ${COVERAGE_CMD} setup.py --name --version --license 2>${_TEST_ROOT_DIR}/${ERR_FILE} | grep --count -oE "^([0-9].){2}[0-9](a0)?$" 2>${_TEST_ROOT_DIR}/${ERR_FILE} || printf "%s\n" "FAIL: Setup.py version check failed" 1>&2 ;} 2>${_TEST_ROOT_DIR}/${ERR_FILE} >> ${_TEST_ROOT_DIR}/${LOG_FILE} & : ;
+{ "${COVERAGE_CMD} setup.py --name --version --license" 2>"${_TEST_ROOT_DIR}"/"${ERR_FILE}" | grep --count -oE "^([0-9].){2}[0-9](a0)?$" 2>"${_TEST_ROOT_DIR}"/"${ERR_FILE}" || printf "%s\n" "FAIL: Setup.py version check failed" 1>&2 ;} 2>"${_TEST_ROOT_DIR}"/"${ERR_FILE}" >> "${_TEST_ROOT_DIR}"/"${LOG_FILE}" & : ;
 
 wait ;
-printf "%s\n" "End of Setup.py Test one-shot" >> ${_TEST_ROOT_DIR}/${LOG_FILE} ; wait ;
-printf "%s\n" "" >> ${_TEST_ROOT_DIR}/${LOG_FILE} ; wait ;
+printf "%s\n\n" "End of Setup.py Test one-shot" >> "${_TEST_ROOT_DIR}"/"${LOG_FILE}" ; wait ;
 
 # sloppy workaround for now
 if [[ ( -x $(command -v python3) ) ]] ; then
-	$(command -v python3) -m coverage combine 2>${_TEST_ROOT_DIR}/${ERR_FILE} || EXIT_CODE=2 ;
-	$(command -v python3) -m coverage xml --include=setup.py,multicast -o ${_TEST_ROOT_DIR}/test-reports/coverage_setup.xml || EXIT_CODE=2 ;
+	$(command -v python3) -m coverage combine 2>"${_TEST_ROOT_DIR}"/"${ERR_FILE}" || EXIT_CODE=2 ;
+	$(command -v python3) -m coverage xml --include=setup.py,multicast -o "${_TEST_ROOT_DIR}"/test-reports/coverage_setup.xml || EXIT_CODE=2 ;
 elif [[ ( -x $(command -v coverage) ) ]] ; then
-	$(command -v coverage) combine 2>${_TEST_ROOT_DIR}/${ERR_FILE} || EXIT_CODE=2 ;
-	$(command -v coverage) xml --include=setup.py,multicast -o ${_TEST_ROOT_DIR}/test-reports/coverage_setup.xml || EXIT_CODE=2 ;
+	$(command -v coverage) combine 2>"${_TEST_ROOT_DIR}"/"${ERR_FILE}" || EXIT_CODE=2 ;
+	$(command -v coverage) xml --include=setup.py,multicast -o "${_TEST_ROOT_DIR}"/test-reports/coverage_setup.xml || EXIT_CODE=2 ;
```

Comment:
:thinking: walk me through the repeated steps, I see only 1 combine and one xm here at 194-195, did I just miss something?

Reply:
so you're suggesting something like this:

```python
COV_ARGS=""
COV_CMD=""

if [[ ( -x $(command -v python3) ) ]] ; then
	 COV_ARGS="-m coverage"
	 COV_CMD=$(command -v python3)
elif [[ ( -x $(command -v coverage) ) ]] ; then
	 COV_ARGS=""
	 COV_CMD=$(command -v coverage)
fi

# ignore to allow COV_ARGS to expand
# shellcheck disable=SC2086
"${COV_CMD}" $COV_ARGS combine 2>"${_TEST_ROOT_DIR}"/"${ERR_FILE}" || EXIT_CODE=2 ;
# shellcheck disable=SC2086
"${COV_CMD}" $COV_ARGS xml --include=setup.py,multicast -o "${_TEST_ROOT_DIR}"/test-reports/coverage_setup.xml || EXIT_CODE=2 ;
```

While I see how this might align with DRY principals it seems to violate AHA principles by increasing overall complexity and LoC (size). Did I miss something?